clang-tools-extra/test/clang-tidy/checkers/performance/avoid-endl.cpp

```diff
@@ -225,3 +225,14 @@ void bad_custom_stream() {
   // CHECK-MESSAGES: :[[@LINE-1]]:13: warning: do not use 'std::endl' with streams; use '\n' instead [performance-avoid-endl]
   // CHECK-FIXES: logger << '\n';
 }
+
+namespace gh107859 {
+
+#define ENDL std::endl;
+
+void bad_macro() {
+  std::cout << ENDL;
```

Comment:
Is `CHECK-FIXES` intentionally not checked?

Reply:
it doesn't have auto fixes

Reply:
But a replacement is always added, isn't it? What am I missing?

Reply:
I am also confused. since I don't find auto fixes even replacement is emitted.
Maybe it is due to invalid location.
Do you think we should explicit avoid this replacement?

Reply:
Yes, for a macro, it's probably better to not suggest a replacement. It would contradict the user's original intention.